From 1fa6cf16064fd08b8789ee02aae9abfa577c2fe0 Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Thu, 1 Oct 2020 19:59:56 +0200
Subject: [PATCH] Don't rely on clang include and binary copies

The qtcreator build system contains a build target
disabled by default which copies the clang include
files and binaries into the qtcreator libexec subdir.

Instead, we'll use the openSUSE's clang locations
directly. This should fix boo#1176888 and work around
https://bugreports.qt.io/browse/QTCREATORBUG-24583 and
https://bugreports.qt.io/browse/QTCREATORBUG-21972.
---
 src/libs/clangsupport/CMakeLists.txt | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/src/libs/clangsupport/CMakeLists.txt b/src/libs/clangsupport/CMakeLists.txt
index dcaa4647..57dd24bd 100644
--- a/src/libs/clangsupport/CMakeLists.txt
+++ b/src/libs/clangsupport/CMakeLists.txt
@@ -5,8 +5,8 @@ add_qtc_library(ClangSupport
   PUBLIC_DEPENDS Utils Sqlite Qt5::Core Qt5::Network
   PUBLIC_DEFINES
     CLANG_VERSION="${CLANG_VERSION}"
-    CLANG_INCLUDE_DIR="${IDE_LIBEXEC_PATH}/clang/lib/clang/${CLANG_VERSION}/include"
-    CLANG_BINDIR="${IDE_LIBEXEC_PATH}/clang/bin"
+    CLANG_INCLUDE_DIR="${LLVM_LIBRARY_DIR}/clang/${CLANG_VERSION}/include"
+    CLANG_BINDIR="${LLVM_LIBRARY_DIR}/clang/${CLANG_VERSION}/bin"
   DEFINES CLANGSUPPORT_BUILD_LIB
   PUBLIC_INCLUDES
     "${CMAKE_CURRENT_LIST_DIR}"
@@ -146,18 +146,3 @@ add_qtc_library(ClangSupport
 if (NOT TARGET libclang)
   return()
 endif()
-
-# For the developer build directory
-qtc_copy_to_builddir(copy_clang_to_builddir
-  DIRECTORIES "${LLVM_LIBRARY_DIR}/clang/${CLANG_VERSION}/include"
-  DESTINATION "${IDE_LIBEXEC_PATH}/clang/lib/clang/${CLANG_VERSION}/include"
-)
-
-foreach(executable clang clang-cl clangd clang-tidy clazy-standalone)
-  if (EXISTS "${LLVM_TOOLS_BINARY_DIR}/${executable}${CMAKE_EXECUTABLE_SUFFIX}")
-    qtc_copy_to_builddir(copy_clang_${executable}_to_builddir
-      FILES "${LLVM_TOOLS_BINARY_DIR}/${executable}${CMAKE_EXECUTABLE_SUFFIX}"
-      DESTINATION "${IDE_LIBEXEC_PATH}/clang/bin/"
-    )
-  endif()
-endforeach()
-- 
2.29.2

